### ISSUE TITLE
Error 204

### ISSUE THREAD

**Issue by hpsilva** [oanda/oandapy#27]
Hi,

Server is returning a **204 Error**, which leads to **No JSON object could be decoded** on the following request as below:

`instrument = 'AU200_AUD'
start = datetime.datetime(2010, 4, 1, 16, 0, 1)
end = datetime.datetime(2010, 4, 5, 3, 20, 1, 1)

candles = oanda.get_history( instrument = instrument,
                            granularity = 'M1',
                            candleFormat = 'midpoint',
                            alignmentTimezone = 'GMT',
                            start = start.isoformat(),
                            end = end.isoformat())`

Any idea on why the server returns such error at this specific dates ?

Thanks

**Comment by hpsilva** [comment]
It seems like there a couple more dates with the same issue on this instrument alone, i.e.:
from start dates between datetime.datetime(2011, 6, 27, 0, 0, 1) up to  datetime.datetime(2011, 6, 27, 0, 0, 1)

I wonder if this are empty entries on the server, or whether i am doing something wrong here

**Comment by hootnot** [comment]
I tested with 2011 instead of 2010. Works fine. I guess OANDA has to take a look server side. The query with your parameters simply returns as content nothing meaning, "". This is not valid JSON, so json.loads gives the error.
The server should return some "No data response" in valid JSON .

**Comment by hpsilva** [comment]
@hootnot Indeed! Based on W3C i have implemented an exception to circumvent the 204 error, but by staying on the same dates or even advancing is of no help when the server returns the error on consecutive date, leading to holes in data.

Quite curiously this do not happen if 'end' date is dismissed, and instead the request is provided with the number of observations ('count' field) to download. 
So seems like data on server is fine, but that request type is raising errors for reasons i can't figure out.

I am trying to work out a historyDownloader to pull from server the entire instrument history. It could be interesting to see something like that implemented in oandapy or oanda-trading-environment!

**Comment by hootnot** [comment]
@OANDA : could you please let the community know if OANDA is still committed to support **oandapy** ?

**Comment by Peque** [comment]
@hootnot It seems not... :unamused:
